sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Misc/MiscellaneousTests.cs

```diff
@@ -0,0 +1,244 @@
+        /// NOTE: the utility analyzers won't run because the required analysis settings are not set
+        /// by the test.
+        /// </summary>
+        private IEnumerable<ExecutionResult> ExecuteAllAnalyzers(Compilation compilation, IEnumerable<Type> analyzerTypes, int analysisTimeoutInMs)
+        {
+            // Run an initial analysis so any startup costs aren't attributed to the first analyzer to be tested
+            ExecuteAnalyzerWithTimeout(compilation, CreateAnalyzer(analyzerTypes.First()), 5000);
+
+            var executionResults = new List<ExecutionResult>();
+            var counter = 0;
+            var timer = new Stopwatch();
+
+            // Executing one analyzer at a time (running them in parallel can skew the results)
+            foreach (var analyzerType in analyzerTypes)
+            {
+                var analyzer = CreateAnalyzer(analyzerType);
+
+                // Write to the DEBUG window so we can see what's happening in the output window in the IDE
+                counter++;
+                Debug.WriteLine($"{counter}: {analyzerType.FullName}");
+
+                timer.Restart();
+                var succeeded = ExecuteAnalyzerWithTimeout(compilation, analyzer, analysisTimeoutInMs);
+                timer.Stop();
+
+                var newExecutionResult = new ExecutionResult(analyzerType, timer.ElapsedMilliseconds, !succeeded);
+                executionResults.Add(newExecutionResult);
+
+                if (!succeeded)
+                {
+                    Debug.WriteLine($"   -> TIMED OUT");
+                }
+            }
+
+            DumpExecutionResults(executionResults);
+            return executionResults;
+        }
+
+        private Compilation GetEFCompilation()
+        {
+            var solutionBuilder = SolutionBuilder.CreateSolutionFromPaths(
+                new string[] { @"TestCases\Misc\Bug2474_20181005212624_InitialCreate.cs" },
+                additionalReferences: GetReferencesNetCore("2.0.0"));
+
+            var compilation = solutionBuilder.Compile(new CSharpParseOptions(LanguageVersion.Latest)).Single();
+            return compilation;
+        }
+
+        private static IEnumerable<MetadataReference> GetReferencesNetCore(string entityFrameworkVersion) =>
+            Enumerable.Empty<MetadataReference>()
+                .Concat(FrameworkMetadataReference.Netstandard)
+                .Concat(NuGetMetadataReference.MicrosoftEntityFrameworkCoreSqlServer(entityFrameworkVersion))
+                .Concat(NuGetMetadataReference.MicrosoftEntityFrameworkCoreRelational(entityFrameworkVersion));
+
+        private Type[] GetSonarAnalyzerTypes(Assembly assembly)
+        {
+            var allAnalyzerTypes = assembly.GetTypes()
+                .Where(IsSonarAnalyzer).ToArray();
+
+            Log($"Analyzer count: {allAnalyzerTypes.Length}");
+
+            return allAnalyzerTypes;
+        }
+
+        private bool IsSonarAnalyzer(Type type) =>
```

Comment:
Yes, it could be. I'm not really a fan of local functions; I don't find that they really help readability or maintainability.

I've made `IsDerivedFromSonarDiagnosticAnalyzer` a local function.